lib/Sema/TypeCheckConcurrency.h

```diff
@@ -436,6 +437,38 @@ namespace detail {
   };
 }
 
+/// Diagnose any non-Sendable types that occur within the given type, using
+/// the given diagnostic.
+///
+/// \returns \c true if any errors were produced, \c false if no diagnostics or
+/// only warnings and notes were produced or if a decl contains a sending
+/// parameter or result
+template <typename... DiagArgs>
+bool diagnoseNonSendableTypesWithSendingCheck(
+    ValueDecl *decl, Type type, SendableCheckContext fromContext,
+    Type derivedConformance, SourceLoc typeLoc, SourceLoc diagnoseLoc,
+    Diag<Type, DiagArgs...> diag,
+    typename detail::Identity<DiagArgs>::type... diagArgs) {
+  if (auto param = dyn_cast<ParamDecl>(decl)) {
+    if (param->isSending()) {
+      return false;
+    }
+  }
+  if (auto *func = dyn_cast<FuncDecl>(decl)) {
+    if (func->hasSendingResult())
+      return false;
+  }
+  if (auto *subscript = dyn_cast<SubscriptDecl>(decl)) {
+    if (auto resultTypeRepr = subscript->getResultTypeRepr())
+      if (isa<SendingTypeRepr>(resultTypeRepr))
```

Comment:
You can simplify this down to `isa_and_nonnull<SendingTypeRepr>(subscript->getResultTypeRepr())`

Reply:
@xedin 
Thank you! I fixed it.
https://github.com/swiftlang/swift/pull/78601/commits/55e8d5579ffa2873d240daaf4558a345f02fad1c